apps/condo/domains/user/schema/SupportMessagesForwardingService.js

```diff
@@ -0,0 +1,123 @@
+const { MESSAGE_FORWARDED_TO_SUPPORT } = require('@condo/domains/notification/constants/constants')
+const { SUPPORT_EMAIL } = require('@condo/domains/common/constants/requisites')
+const { get } = require('lodash')
+const { LOCALES } = require('@condo/domains/common/constants/locale')
+const { Resident } = require('@condo/domains/resident/utils/serverSchema')
+const { Organization } = require('@condo/domains/organization/utils/serverSchema')
+const FileAdapter = require('@condo/domains/common/utils/fileAdapter')
+const { v4: uuid } = require('uuid')
+const dayjs = require('dayjs')
+
+const FORWARDED_EMAILS_ATTACHMENTS_FILE_FOLDER_NAME = 'forwarded-emails-attachments'
+const adapter = new FileAdapter(FORWARDED_EMAILS_ATTACHMENTS_FILE_FOLDER_NAME)
+
+const SupportMessagesForwardingService = new GQLCustomSchema('SupportMessagesForwardingService', {
+    types: [
+        {
+            access: true,
+            type: `enum SupportMessagesForwardingLang { ${Object.keys(LOCALES).join(' ')} }`,
+        },
+        {
+            access: true,
+            type: 'input SupportMessagesForwardingInput { dv: Int!, sender: JSON!, text: String!, email: String, attachments: [Upload], os: String!, appVersion: String!, lang: SupportMessagesForwardingLang!, meta: JSON! }',
+        },
+        {
+            access: true,
+            type: 'type SupportMessagesForwardingOutput { id: String!, status: String! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canSupportMessagesForwarding,
+            schema: 'supportMessagesForwarding(data: SupportMessagesForwardingInput!): SupportMessagesForwardingOutput',
+            resolver: async (parent, args, context, info, extra = {}) => {
+                const { data } = args
+                const { dv, sender, text, email: emailFrom, attachments = [], os, appVersion, lang, meta } = data
+
+                const user = get(context, ['req', 'user'])
+
+                const attachmentsData = await Promise.all(attachments)
+
+                const filesPromises = attachmentsData.map((result) => {
+                    const { filename: originalFilename, mimetype, encoding, createReadStream } = result
+                    const stream = createReadStream()
+                    return adapter.save({
+                        stream,
+                        id: `${dayjs().format('YYYY-MM-DD_HH-mm-ss')}_${uuid()}`,
+                        filename: originalFilename,
+                    }).then(({ filename, id, ...restSaveData }) => {
+                        const ret = {
+                            encoding,
+                            filename,
+                            id,
+                            mimetype,
+                            originalFilename,
+                            publicUrl: adapter.publicUrl({ filename }),
+                        }
+
+                        if (FileAdapter.isLocal(adapter)) {
```

Comment:
It should work out of the box. why do you need to patch a publicUrl?

Reply:
I'd like to get files from a local disk. So, I need a full path with `.../__media/...` instead of `.../media/...`.

Reply:
Probably, it's an error. You should have two options, one for a local path and the second for a URL. Probably, it's a configuration error or local middleware error. The publicUrl should work well out of the box

Reply:
I think this is no error. The `publicUrl` field contains real URL, which should be accessed via express (http).
<img width="1160" alt="image" src="https://user-images.githubusercontent.com/25384290/154619932-6bda2fea-fa49-4d84-9061-ae4610934bdb.png">
But these URLs are used by worker on the same host, so we do not need to access them via http. I decided to use local path to access the files.
Anyway, even if we keep the original `publicUrl`, we have to add the application http host at the beginning of this url.

Reply:
Also, I think it's a good idea to rename `FileAdapter` to `FileAdaptersFactory` to clarify the purpose of the class.